### PR TITLE
gh-115538: Emit warning when use bool as fd in _io.WindowsConsoleIO

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-115538

### CODE REVIEW

**Review by serhiy-storchaka — COMMENTED** [review]
Could you please add explicit tests for passing boolean to _WindowsConsoleIO constructor?

There are tests that pass 0 and 1 to _WindowsConsoleIO constructor in Lib/test/test_winconsoleio.py. You can add new tests for booleans there.

**Comment by aisk** [comment]
Thank you for the review, test case added. The Tests / CIFuzz (memory) kept failing, but I think it's not caused by this change. Other changes in the main branch code have the same issue.

File: Lib/test/test_winconsoleio.py [hunk 1]
```diff
@@ -68,6 +69,18 @@ def test_open_fd(self):
             f.close()
             f.close()
 
+        with warnings.catch_warnings():
+            warnings.simplefilter("error", RuntimeWarning)
```
Comment (serhiy-storchaka) [inline Lib/test/test_winconsoleio.py]: It should not be needed if `assertRaises` is used.

File: Lib/test/test_winconsoleio.py [hunk 2]
```diff
@@ -68,6 +69,18 @@ def test_open_fd(self):
...
+            try:
+                with self.assertRaises(RuntimeWarning):
+                    f = ConIO(False)
```
Comment (serhiy-storchaka) [inline Lib/test/test_winconsoleio.py]: Opened file should be closed.

File: Lib/test/test_winconsoleio.py [hunk 3]
```diff
@@ -68,6 +69,18 @@ def test_open_fd(self):
...
+                with self.assertRaises(RuntimeWarning):
+                    f = ConIO(True, mode='w')
+            except ValueError:
+                # cannot open console because it's not a real console
+                pass
```
Comment (serhiy-storchaka) [inline Lib/test/test_winconsoleio.py]: I think that it is better to run these tests only after we checked that the code works with corresponding integer argument. Add them in the "else" branches above.

File: Lib/test/test_winconsoleio.py (hunk 1 above)
Comment (aisk) [reply Lib/test/test_winconsoleio.py]: I tested and found that these two lines are needed. However, I noticed that there is an `assertWarns` function which does not require these two lines.

The original code is copied from existing codes, so maybe we should update them later.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**